cmds/core/pci/pci.go

```diff
@@ -109,16 +109,72 @@ func registers(d pci.Devices, cmds ...string) {
 
 	}
 }
+
+func readsize(want int) {
+	if want > readSize {
+		readSize = want
+	}
+}
+
+func dumpsize(want int) {
+	readsize(want)
+	if want > dumpSize {
+		dumpSize = want
+	}
+
+}
+
+func init() {
+	args := os.Args
+	os.Args = nil
+	// PCI command arguments and Go arguments are not very compatible.
+	// Look for certain patterns, and on match, discard them and note them.
+	// Consider doing this via regexp but, in the end, it's unlikely to be easier.
+	for _, a := range args {
+		switch a {
+		case "-v":
+			readsize(48)
+			verbose++
+		case "-vv":
+			readsize(256)
+			verbose = 2
+		case "-vvv":
+			readsize(4096)
+			verbose = 3
+		case "-x":
+			switch dumpSize {
+			case 0:
+				dumpsize(48)
+			case 1:
+				dumpsize(256)
+			case 2:
+				dumpsize(4096)
+			default:
+				dumpsize(4096)
+			}
+		case "-xxx":
+			dumpsize(256)
+		case "-xxxx":
```

Comment:
It's not the most elegant parsing of parameters. Isn't there a getopt in go?
What about -vvvxxx? Do we really want that to be different from -vvv -xxx ?

Reply:
it's all we've got. Sorry.